### PR TITLE
[RESTRO 2] Setup express server

### PR DESCRIPTION
**Pull request by kavitamakariye**
@Hdonge please review the PR.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+
+const app = express();
+const port = 2000;
```
Comment (Hdonge) [inline index.js]: Create a file name `config.json` and move configurations parameters there. Use `nconf` npm package to read configs from config file.
Configs can be created like

```
"environmant":{
      "production":{
                 "port":  3000
    },
      "development":{
                 "port":  5000
     }
}
```

Comment (Hdonge) [reply index.js]: https://github.com/Hdonge/nodejs-app/blob/master/app/lib/config.js refer this

File: index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+app.get('/', (req,res)=>{
+    res.send('Hello!!');
```
Comment (Hdonge) [inline index.js]: It can be `res.send(200,{Status:'ok'});`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+})
+
+app.listen(port, ()=> {
+    console.log(`Listening on port ${port}!!`);
```
Comment (Hdonge) [inline index.js]: `console.log` should not be the part of the real time app. Use `log4js` npm package instead.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  "name": "restro-booking-server",
+  "version": "1.0.0",
+  "description": "This is server side application for Smart Online Restaurant Booking System",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/kavitamakariye/restro-booking-server.git"
```
Comment (Hdonge) [inline package.json]: Should be upstream url everywhere

Comment (Hdonge) [inline package.json]: Should be upstream url everywhere

File: index.js (hunk 1 above)
Comment (Hdonge) [reply index.js]: https://github.com/Hdonge/nodejs-app/blob/master/app/lib/server.js#L186

File: index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
```
Comment (Hdonge) [inline index.js]: `main` file of the application should be `index.js`.  But this is the code represent server is setup. Move this code inside `server.js` file and invoke that function from here. 

refer this https://github.com/Hdonge/nodejs-app/blob/master/app/index.js

**Review by Hdonge — CHANGES_REQUESTED** [review]
Some changes requested. Majorly infrastructure setup related changes required in early stage of application developments

**Comment by kavitamakariye** [comment]
Made all the changes, please review it.

File: config.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+    "environment":{
+    "production":{
+               "port":  3000
```
Comment (Hdonge) [inline config.json:4]: After doing changes in any `.js Or . json ` please do indentation properly. Ctrl+shift+I

File: server.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+//Required express, nconf and log4js
```
Comment (Hdonge) [inline server.js:1]: My intention was not to rename index.js to server.js but create a separate server file in that server related code can be moved and invoke that from index file

File: server.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+const logger = log4js.getLogger();
+logger.level = 'debug'
+
+//Get port from the conifg.json file
+nconf.env();
+nconf.file('./config.json')
+var port = nconf.get('environment:development:port');
+
+//Create instance of express
+const app = express();
```
Comment (Hdonge) [inline server.js:16]: Keep all the variable declarations at the top of the file. This is the part of code styling and make easy to debug.

File: server.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+app.get('/', (req,res)=>{
+    res.send(200,{Status:'ok'});
```
Comment (Hdonge) [inline server.js:19]: Object property should always be in camel case.  Should be `status` here

File: server.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+//Required express, nconf and log4js
+const express = require('express');
+const nconf = require('nconf');
+const log4js = require('log4js');
```
Comment (Hdonge) [inline server.js:4]: Configuration setup for log4js and nconf should be in separate module and those should be consume here.  
I will do in next pr. Will separate out everything.

**Review by Hdonge — APPROVED** [review]
This is initial phase of application development where infrastructure setup is more important. Newly given reviews are mostly related to code styling and module creations. I will try to do that in next pr. You can follow them letter. 
Rest the task wise changes looks good. Approving it.

**Comment by Hdonge** [comment]
@kavitamakariye I am merging this pr and will create a task for me to do some changes related to infra setup.

**Pull request merged**